smolyay/basis.py

```diff
@@ -135,6 +136,11 @@ def domain(self):
         """numpy.ndarray: Domain the sample points come from."""
         return numpy.array([-1, 1])
 
+    @property
```

Comment:
This doesn't need to be part of the public API. I would make it a private class variable of the base class that defaults to `False`, and then override it in a deriving class that needs it to be true. This would go in `BasisFunction`
```py
    _is_complex = False
```